zgw_consumers/client.py

```diff
@@ -1,128 +1,99 @@
 import logging
-import warnings
-from dataclasses import dataclass
-from typing import IO, Any, Dict, List, Optional, Union
-from urllib.parse import urljoin
+from dataclasses import dataclass, field
+from typing import Any, TypeVar
 
-from django.utils.module_loading import import_string
+from requests.auth import AuthBase
+from requests.models import PreparedRequest
+from zds_client import ClientAuth
 
-import yaml
-import zds_client
-from zds_client import Client
-from zds_client.oas import schema_fetcher
+from zgw_consumers.constants import AuthTypes
+from zgw_consumers.models import Service
 
-from .settings import get_setting
+# For backwards compatibility:
+from .legacy.client import UnknownService, ZGWClient, get_client_class, load_schema_file
```

Comment:
`get_client_class` and `load_schema_file` are not documented as public API on https://zgw-consumers.readthedocs.io/en/latest/, so we might even get away with not exposing these here.

Looking at https://zgw-consumers.readthedocs.io/en/latest/quickstart.html#usage, we might even skip the compatibility layer alltogether, as only the `Service` model is really considered public API. Of course, the changelog can mention that these names are moved to `legacy` and tell you which imports to update to accomodate users.